scripts/west_commands/runners/blackmagicprobe.py

```diff
@@ -150,16 +152,16 @@ def do_add_parser(cls, parser):
                             help='Assert SRST during connect? (default: no)')
 
     def bmp_flash(self, command, **kwargs):
-        if self.elf_file is None:
-            raise ValueError('Cannot debug; elf file is missing')
+        if self.hex_file is None:
+            raise ValueError('Cannot flash; hex file is missing')
```

Comment:
Building the HEX file is optional (BUILD_OUTPUT_HEX), so at minimum this should still take the elf file if the hex is not specified, but also the output of gdb when flashing with the elf file is IMO nicer when using the elf than when using the hex file (it shows the section names rather than `Loading section .sec1...`.

How about:
- if hex file is not specified, use elf file
- if the hex and elf basename are the same, use the elf
- otherwise use the hex

that way the current behavior is unchanged unless there's a hint that the hex file has been further processed (by having a different name)

Reply:
Thanks for review. I've added this fallback.